doc/vhpidirect/examples/arrays.rst

```diff
@@ -0,0 +1,56 @@
+.. program:: ghdl
+.. _COSIM:VHPIDIRECT:Examples:arrays:
+
+Arrays
+######
+
+.. _COSIM:VHPIDIRECT:Examples:arrays:logicvectors:
+
+:cosimtree:`logicvectors <vhpidirect/arrays/logicvectors>`
+**************************************************************
+
+Commonly signals in VHDL are of a logic type or a vector thereof (``std_logic`` and ``std_logic_vector``), coming from the ``ieee.std_logic_1164.*`` package.
+These types can hold values other than high and low (``1`` and ``0``) and are enumerated as:
+
+0. 'U' 
+1. 'X' 
+2. '0' 
+3. '1' 
+4. 'Z' 
+5. 'W' 
+6. 'L' 
+7. 'H' 
+8. '-' 
+
+Because the number of enumeration values is less than 256, logic values are transported in 8 bit words (a ``char`` type in C) (as mentioned in :ref:`Restrictions_on_foreign_declarations`).
+Logic vectors, of a bounded size, are easily created in C and passed to VHDL as ``char[]``. VHDL receives pointers as ``access`` types, in this case an access of a subtype of std_logic_vector.
+
+In this example two declarations make handling logic values in C a bit easier:
+
+- Providing logic values in C as their enumeration numbers is simplified with the use of a matching enumeration, ``HDL_LOGIC_STATES``. 
+- Printing out a logic value's associated character is also simplified with the ``const char HDL_LOGIC_CHAR[]`` declaration.
+
+This example builds on the matrices example (:ref:`COSIM:VHPIDIRECT:Examples:quickstart:matrices:`). Foreign subprograms are declared that enable receiving the size of two different logic vectors as well as the vectors themselves from C. There is only one subprogram to get the size of both C arrays, and it takes in an integer to determine which array's size gets returned. 
+
+.. HINT::
+  The ``getLogicVecSize`` in VHDL is declared as receiving a ``boolean`` argument. In C the function is declared to receive an ``char`` argument. The VHDL booleans ``false`` and ``true`` are enumerations, and have integer values, ``0`` and ``1`` respectively. As with the logic values, the boolean enumerations use fewer than 8 bits, so the single byte in C's ``char`` variable receives the VHDL enumeration correctly.
+
+
+The two vectors are populated with logic values in different ways:
+
+- LogicVectorA's indices are manually filled with enumeration values from HDL_LOGIC_STATES.
+
+  - .. code-block:: C
+
+        logic_vec_A[0] = HDL_U;
+
+- LogicVectorB's indices are filled with an integer value.
+
+  - .. code-block:: C
+
+        for(int i = 0; i < SIZE_LOGIC_VEC_B; i++){
+          logic_vec_B[i] = 8-i;
+        }
+
+.. ATTENTION::
```

Comment:
Maybe merge this ATTENTION with the HINT above. I'm not sure about the location (here or there).

Reply:
I really like where it is. Feels a little like it is floating, but it is right underneath the code detail where _integer values are given to ``char`` variables in C which are intended to be read as VHDL logic values_

Reply:
Then, merge the HINT into this ATTENTION and leave everything below. Does it make sense?

Reply:
I think I am just being pedantic about using the HINT and ATTENTION differently. The HINT about the boolean isn't a dangerous point.. It is correct code that cannot really go wrong, it just isn't explained before then, not that the explanation is even a sore point.

The ATTENTION is about a dangerous point. Those values being assigned to the char aren't ensured to be [0, 8]... 

![image](https://user-images.githubusercontent.com/28019482/79669479-0ec75280-81bc-11ea-8b20-0e6d1dd23e74.png)

If you really want it merged I'll do it. But I think they deserve the separate colours and different attention.

Reply:
I mean, I really wanted the attention to be a part of the LogicVectorB sub list, but it is also good to be making a general warning.

Reply:
I beg your pardon. You are absolutely correct. I understood that we were talking about the same `char`... 🤦 

It has absolutely nothing to do! One is a boolean, the other one is a std_logic. Both are enumerations and both are mapped as chars. However, booleans have range [0,1] and std_logic have range [0,8].